build.gradle.kts

```diff
@@ -3,160 +3,51 @@
+                implementation(libs.kotlinxBenchmark.runtime)
+                implementation(libs.ashley)
+                implementation(libs.artemisOdb)
             }
         }
-        val jsMain by getting
-        val jsTest by getting
-        val nativeMain by getting
-        val nativeTest by getting
     }
 }
 
-val javadocJar by tasks.registering(Jar::class) {
-    archiveClassifier.set("javadoc")
+tasks.javadocJar {
     from(tasks.dokkaHtml)
 }
-
-publishing {
-    repositories {
-        maven {
-            url = if (project.version.toString().endsWith("SNAPSHOT")) {
-                uri("https://s01.oss.sonatype.org/content/repositories/snapshots/")
-            } else {
-                uri("https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/")
-            }
-
-            credentials {
-                username = System.getenv("OSSRH_USERNAME")
-                password = System.getenv("OSSRH_TOKEN")
-            }
-        }
-    }
-
-    publications {
-        val kotlinMultiplatform by getting(MavenPublication::class) {
-            // we need to keep this block up here because
-            // otherwise the different target folders like js/jvm/native are not created
-            version = project.version.toString()
-            groupId = project.group.toString()
-            artifactId = "Fleks"
-        }
-    }
-
-    publications.forEach {
-        if (it !is MavenPublication) {
-            return@forEach
-        }
-
-        // We need to add the javadocJar to every publication
-        // because otherwise maven is complaining.
-        // It is not sufficient to only have it in the "root" folder.
-        it.artifact(javadocJar)
-
-        // pom information needs to be specified per publication
-        // because otherwise maven will complain again that
-        // information like license, developer or url are missing.
-        it.pom {
-            name.set("Fleks")
-            description.set("A lightweight entity component system written in Kotlin.")
-            url.set("https://github.com/Quillraven/Fleks")
-
-            scm {
-                connection.set("scm:git:git@github.com:quillraven/fleks.git")
-                developerConnection.set("scm:git:git@github.com:quillraven/fleks.git")
-                url.set("https://github.com/quillraven/fleks/")
-            }
-
-            licenses {
-                license {
-                    name.set("MIT License")
-                    url.set("https://opensource.org/licenses/MIT")
-                }
-            }
-
-            developers {
-                developer {
-                    id.set("Quillraven")
-                    name.set("Simon Klausner")
-                    email.set("quillraven@gmail.com")
-                }
-            }
-        }
-
-        signing {
-            useInMemoryPgpKeys(System.getenv("SIGNING_KEY"), System.getenv("SIGNING_PASSWORD"))
-            sign(it)
-        }
-    }
-}
```

Comment:
Publishing configuration has been moved to `buildSrc/src/main/kotlin/buildsrc/conventions/publishing.gradle.kts`. This helps keep the `build.gradle.kts` of subprojects smaller and focused, and makes it easier to share the same config in any new subprojects.